sentinelone/komand_sentinelone/util/api.py

```diff
@@ -9,23 +9,34 @@ def __init__(self, url, make_token_header):
         self.url = url
         self.token_header = make_token_header
 
-    def search_agents(self, agent_details: str, agent_active: bool = True, results_length: int = 0) -> list:
+    def search_agents(self, agent_details: str, agent_active: bool = True, case_sensitive: bool = True, results_length: int = 0) -> list:
         results = []
         if agent_details:
             for search in self.__get_searches(agent_details):
-                endpoint = f"{self.url}web/api/v2.0/agents?{search}={agent_details}"
-                output = requests.get(endpoint, headers=self.token_header)
+                agents = [agent_details]
 
-                if output.status_code is 200 and output.json()["pagination"]["totalItems"] >= 1:
-                    results.append(output.json()['data'][0])
-                    if results_length:
-                        if len(results) >= results_length:
-                            break
+                # Normalize casing if specified
+                if not case_sensitive:
+                    if search == "computerName":
+                        agents = [agent_details.lower(), agent_details.upper()]
+                    if search == "uuid":
+                        agents = [agent_details.lower()]
+
+                for agent in agents:
+                    endpoint = f"{self.url}web/api/v2.0/agents?{search}={agent}"
+                    output = requests.get(endpoint, headers=self.token_header)
+
+                    if output.status_code is 200 and output.json()["pagination"]["totalItems"] >= 1:
+                        results.append(output.json()['data'][0])
+
+                if results_length:
+                    if len(results) >= results_length:
+                        return self.clean_results(results)
```

Comment:
I know we talked about how to handle this in a meeting the other day but let me know if you want to use a different approach to the case sensitivity issue -- open to any and all suggestions!